File: android-lib/dmlib/src/main/java/com/damai/core/DMPage.java
package com.damai.core;

import java.util.List;

/**
 * 用来描述一个分页列表
 * @author Randy
 *
 * @param <T>
 */
public class DMPage<T> {
    List<T> list;
    /**
     * 当前分页，1开始
     */
    int page;
    /**
     * 总分页
     */
    int total;


    public List<T> getList() {
        return list;
    }

    public int getPage() {
        return page;
    }

    public int getTotal() {
        return total;
    }

    public boolean isLast(){
        return page >= total;
    }

    public boolean isFirst(){
        return page <= 1;
    }

}

File: mod_ios/dmlib/dmlib/http/DMPage.m
//
//  Page.m
//  libs
//
//  Created by randy ren on 16/1/7.
//  Copyright © 2016年 damai. All rights reserved.
//

#import "DMPage.h"

@implementation DMPage
-(BOOL)isLast{
    return _page >= _total;
}
-(BOOL)isFirst{
    return _page <=1;
}
-(NSDictionary*)toJson{
  return @{
           @"page":[NSNumber numberWithInteger: self.page],
           @"total":[NSNumber numberWithInteger: self.total],
           @"result":self.list,
           @"extra":self.extra ? self.extra : [NSNull null]
           };
}
@end
